beacon-chain/sync/subscriber_beacon_blocks.go

```diff
@@ -37,12 +38,13 @@ func (s *Service) beaconBlockSubscriber(ctx context.Context, msg proto.Message)
 		return err
 	}
 
-	// Delete attestations from the block in the pool to avoid inclusion in future block.
-	if err := s.deleteAttsInPool(block.Body().Attestations()); err != nil {
-		log.Debugf("Could not delete attestations in pool: %v", err)
-		return nil
+	if !featureconfig.Get().CorrectlyPruneCanonicalAtts {
```

Comment:
Same feedback, let's move this conditional inside the function

Reply:
This is a different function

Reply:
can we move the correctlyprunecanonicalatts conditional inside of deleteAttsInPool ?